ticketmatch.rb

```diff
@@ -307,7 +311,18 @@ def inspect
 #
 git_commits.keys.sort.each do |ticket|
   if jira_tickets[ticket].nil?
-    puts "** #{ticket.upcase}"
+    begin
+      actual = JSON.parse(%x{curl -X GET -H 'Content-Type: application/json' https://tickets.puppetlabs.com/rest/api/2/issue/#{ticket}})
+      if actual.key?('key') && jira_tickets[actual['key']]
+        jira_tickets.add_alias(actual['key'], ticket)
+        puts "-- #{ticket.upcase} (#{jira_tickets[ticket].state}) [alias of #{actual['key']}]"
+      else
+        puts "** #{ticket.upcase}"
+      end
+    rescue
+      # Failed to look up the ticket.
+      puts "** #{ticket.upcase}"
+    end
```

Comment:
Do you have a copy of the output from our baseline problem that prompted this enhancement?

Reply:
Here are the relevant sections:
```
<><><><><><><><><><>
Note: fetch origin for puppet-agent...
Checking: puppet-agent
from_rev: 6.0.2, to_rev: 6.0.3_release, fix_ver: 6.0.3
...
** PUP-8351
    72b1f6a6  Update bash wrappers to keep LD_LIBRARY_PATH with RHSCL
...
...
----- Git commits in Jira -----
COMMIT TOKENS NOT FOUND IN JIRA (OR NOT WITH FIX VERSION OF puppet-agent 6.0.3)
PUP-8351
```
This commit now evaluates as

Reply:
Did the last half not paste?

Reply:
Thats excellent!